components/ActivityForm/ActivityForm.js

```diff
@@ -0,0 +1,169 @@
+    };
+
+    const response = await fetch("/api/activities", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(activityData),
+    });
+
+    if (response.ok) {
+      setSuccessMessage("Your activity was added successfully!");
+      setErrorMessage("");
+      mutate();
+      event.target.reset();
+    } else {
+      setErrorMessage("Something went wrong. Please try again.");
+      setSuccessMessage("");
+      console.error("Failed to create activity");
+    }
+  }
+
+  if (categoriesLoading) return <p>Loading categories...</p>;
+  if (categoriesError) return <p>Failed to load categories.</p>;
+
+  return (
+    <FormSection>
+      <SectionHeader>
+        <h2>Add your activity</h2>
+        <ToggleButton onClick={() => setIsFormOpen((prev) => !prev)}>
+          {isFormOpen ? "Hide form" : "Show form"}
+        </ToggleButton>
+      </SectionHeader>
+      {successMessage && (
+        <StatusMessage $success>{successMessage}</StatusMessage>
+      )}
+      {errorMessage && <StatusMessage>{errorMessage}</StatusMessage>}
+      <FormContent isOpen={isFormOpen}>
+        <StyledForm onSubmit={handleSubmit}>
+          <StyledFormLabel htmlFor="activity-title">Title:</StyledFormLabel>
+          <StyledFormInput
+            id="activity-title"
+            type="text"
+            name="title"
+            maxLength="50"
+            placeholder="Name your activity (e.g. Kayaking)"
+            required
+          />
+
+          <StyledFormLabel htmlFor="activity-description">
+            Description:
+          </StyledFormLabel>
+          <StyledFormInput
+            id="activity-description"
+            type="text"
+            name="description"
+            maxLength="300"
+            placeholder="Add a short description"
+          />
+
+          <CategoryGroup>
+            <CategoryLegend>Choose categories:</CategoryLegend>
+            <CategoryList>
+              {categories.map((category) => (
+                <CategoryItem key={category._id}>
+                  <CategoryLabel>
+                    <StyledFormInput
+                      type="checkbox"
+                      id="activity-categories"
+                      name="categories"
+                      value={category._id}
+                      multiple
+                    />
+                    {category.name}
+                  </CategoryLabel>
+                </CategoryItem>
+              ))}
+            </CategoryList>
+          </CategoryGroup>
+
+          <StyledFormLabel htmlFor="activity-area">Area:</StyledFormLabel>
+          <StyledFormInput
+            id="activity-area"
+            type="text"
+            name="area"
+            placeholder="e.g. Alps, Black Forest, Lake District"
+          />
+
+          <StyledFormLabel htmlFor="activity-country">Country:</StyledFormLabel>
+          <CountryCombobox
+            id="activity-country"
+            name="country"
```

Comment:
Do you need to pass 'id' and 'name' as props? Or can they be defined in the ContryCombobox component itself?